MoltenVK/MoltenVK/GPUObjects/MVKSync.mm

```diff
@@ -155,6 +160,76 @@
 }
 
 
+#pragma mark -
+#pragma mark MVKEventNative
+
+// Odd == set / Even == reset.
+bool MVKEventNative::isSet() { return _mtlEvent.signaledValue & 1; }
+
+void MVKEventNative::signal(bool status) {
+	if (isSet() != status) {
+		_mtlEvent.signaledValue += 1;
+	}
+}
+
+void MVKEventNative::encodeSignal(id<MTLCommandBuffer> mtlCmdBuff, bool status) {
+	if (isSet() != status) {
+		[mtlCmdBuff encodeSignalEvent: _mtlEvent value: _mtlEvent.signaledValue + 1];
```

Comment:
Another problem: these methods of `MTLCommandBuffer` may only be called when no command encoders are active. (Try running with the Metal validation layer on.) You'll have to end the current encoder before calling these methods. And if we're in the middle of a render pass, you'll have to start it back up again.

Reply:
> Another problem: these methods of `MTLCommandBuffer` may only be called when no command encoders are active. (Try running with the Metal validation layer on.) You'll have to end the current encoder before calling these methods. And if we're in the middle of a render pass, you'll have to start it back up again.

Thanks. Yup...this is covered in `MVKCommandEncoder::signalEvent()`. And yup...I have been running CTS with Metal validation on. Vulkan `vkCmdRe/SetEvent()` singling is outside render pass scope (matching the Metal requirement you mention).

Reply:
I stand corrected. You did indeed end the current encoder there. Also, apparently, `vkCmd{Set,Reset}Event()` cannot be executed inside a render pass.

I also notice that the restriction documented for `-[MTLCommandBuffer encodeSignalEvent:value:]` is not present in the docs at all for `-[MTLCommandBuffer encodeWaitForEvent:value:]`. So I guess I was worried for nothing.

Reply:
Yeah...looks like in both Vulkan & Metal...the event wait can occur within a render pass...but the signalling cannot.